scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,263 @@
+package org.oppia.android.scripts
+
+import java.io.File
+import java.io.FileInputStream
+import org.oppia.android.scripts.proto.FilenameChecks
+import org.oppia.android.scripts.proto.FilenameCheck
+import org.oppia.android.scripts.proto.FileContentChecks
+import org.oppia.android.scripts.proto.FileContentCheck
+import org.oppia.android.scripts.ScriptResultConstants
+import com.google.protobuf.MessageLite
+
+/** Script for ensuring that prohibited file contents and
+ *  file naming patterns are not present in the codebase.
+ */
+class RegexPatternValidationCheck {
+  companion object {
+    @JvmStatic
+    fun main(vararg args: String) {
+      /** path of the repo to be analyzed. */
+      val repoPath = args[0] + "/"
+
+      /** a list of all allowed directories in the repo to be analyzed. */
+      val allowedDirectories = args.drop(1)
+
+      /** a list of all files in the repo to be analyzed. */
+      val searchFiles = RepoFile.collectSearchFiles(
+        repoPath = repoPath,
+        allowedDirectories = allowedDirectories
+      )
+
+      /** check if the repo has any filename failure. */
+      val hasFilenameCheckFailure = retrieveFilenameChecks()
+        .fold(initial = false) { isFailing, check ->
+          val checkFailed = checkProhibitedFileNamePattern(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedFilenameObj = check,
+          )
+          return@fold isFailing || checkFailed
+        }
+
+      /** check if the repo has any file content failure. */
+      val hasFileContentCheckFailure = retrieveFileContentChecks()
+        .fold(initial = false) { isFailing, check ->
+          val checkFailed = checkProhibitedContent(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedContentObj = check,
+          )
+          return@fold isFailing || checkFailed
+        }
+
+      if (hasFilenameCheckFailure || hasFileContentCheckFailure) {
+        throw Exception(ScriptResultConstants.REGEX_CHECKS_FAILED)
+      } else {
+        println(ScriptResultConstants.REGEX_CHECKS_PASSED)
+      }
+
+    }
+
+    /**
+     * Retrieves all filename checks.
+     *
+     * @return a list of all the FilenameChecks
+     */
+    private fun retrieveFilenameChecks(): List<FilenameCheck> {
```

Comment:
Could we move private functions outside companion object?
WDYT?

Reply:
Ah! if we move them outside the companion object then these functions have to be referenced by an instance of the class. This will look comparatively less clean compared to the current approach. These are helper functions that are specifically created for this class, hence it makes sense to reference them directly in the `main` function without involving the reference of any object.

Reply:
This method will only be used in `RegexPatternValidationCheck` class, right? 
Making them static by keeping them inside the companion object doesn't look good to me. 

We usually keep things inside companion object which we want other classes in the same package to access without creating an object of that particular class; example - `CLASS_NAME.FUNCTION()`. 

So, here we don't want any other class to access this method, and we are making it private too, so I didn't see any good case to keep it inside the companion object.

Reply:
Moved them outside the `companion object`. PTAL.